### DISCUSSION TITLE
About teleport limits

### DISCUSSION THREAD

**Discussion by mukitaro** [gravitational/teleport#20760] [Q&A] [answered]
Hi,
Is there any documentation on teleport limits?
ex. The maximum number of roles that can be set for a user.

**Comment by webvictim** [comment, marked as answer]
There is no enforced limitation on the number of roles that a Teleport user can have.

Naturally as a user is given more roles, the process of computing RBAC rules becomes more computationally intensive, so you can expect an increase in time taken to process RBAC rules as the number of roles that a user has increases.

Generally these computations are restricted by the available CPU power that the Teleport auth service has available.

**Reply by mukitaro** [reply]
Hi webvictim

Thank you for providing an answer to my question.
I understood it.
